transports/noise/src/io/framed.rs

```diff
@@ -18,22 +18,18 @@
 // FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
 // DEALINGS IN THE SOFTWARE.
 
-//! This module provides a `Sink` and `Stream` for length-delimited
-//! Noise protocol messages in form of [`NoiseFramed`].
+//! This module provides a `Codec` type implementing the `Encoder` and `Decoder`
+//! traits, making it useful alongside a `asynchronous_codec::Framed` to provide
+//! `Sink` and `Stream` for length-delimited Noise protocol messages.
 
-use crate::io::Output;
+use super::handshake::proto;
 use crate::{protocol::PublicKey, Error};
-use bytes::{Bytes, BytesMut};
-use futures::prelude::*;
-use futures::ready;
-use log::{debug, trace};
-use std::{
-    fmt, io,
-    pin::Pin,
-    task::{Context, Poll},
-};
-
-/// Max. size of a noise message.
+use bytes::{Buf, BufMut, Bytes, BytesMut};
+use log::{debug, error};
+use quick_protobuf::{BytesReader, MessageRead, MessageWrite, Writer};
+use std::io;
+
+/// Max size of a noise message.
```

Comment:
```suggestion
/// Max. size of a noise message.
```

Nit: Opportunity to reduce diff a touch.